src/System.CodeDom/src/System/CodeDom/Compiler/CompilerResults.cs

```diff
@@ -9,31 +9,19 @@ namespace System.CodeDom.Compiler
 {
     public partial class CompilerResults
     {
-        private Assembly _compiledAssembly;
-
         public CompilerResults(TempFileCollection tempFiles)
         {
             TempFiles = tempFiles;
         }
 
         public TempFileCollection TempFiles { get; set; }
 
-        public Assembly CompiledAssembly
-        {
-            get
-            {
-                if (_compiledAssembly == null && PathToAssembly != null)
-                {
-                    _compiledAssembly = Assembly.Load(new AssemblyName() { CodeBase = PathToAssembly });
```

Comment:
Seems it was working fine in full framework, what if we broke something by removing it?

Reply:
We won't ever support loading from CodeBase (/cc @jkotas: https://github.com/dotnet/coreclr/issues/10561)

Reply:
We may want to change this to `_compiledAssembly  = Assembly.LoadFile(PathToAssembly);`

Reply:
Unfortunately
```cs
var t = typeof(Program).Assembly.CodeBase;
Assembly.LoadFile(t);
```

Throws
```
[System.ArgumentException: Absolute path information is required.]
   at System.Security.Util.StringExpressionSet.CreateListFromExpressions(String[] str, Boolean needFullPath)
   at System.Security.Permissions.FileIOPermission.AddPathList(FileIOPermissionAccess access, AccessControlActions control, String[] pathListOrig, Boolean checkForDuplicates, Boolean needFullPath, Boolean copyPathList)
   at System.Security.Permissions.FileIOPermission..ctor(FileIOPermissionAccess access, String path)
   at System.Reflection.Assembly.LoadFile(String path)
   at Program.Main() :line 10
```

This means porting to .NET Core would still throw

Reply:
@hughbe I looked to this issue/PR more and found that @jkotas suggestion works if we use Assembly.Location instead of `Assembly.CodeBase` and talked with @jkotas, turns out `Assembly.CodeBase` is going to become obsolete and `Assembly.Location` is preffered usage. So we prefer to change this row to 
`_compiledAssembly = Assembly.LoadFile(PathToAssembly);` 
instead of removing this implementation.